### ISSUE TITLE
Can't install - could not resolve dependency 'react'

### ISSUE THREAD

**Issue by Ansis100** [mattmilburn/strapi-plugin-menus#112]
Hi! I'm want to implement this plugin but can't install it.

This is the bash log:
```shell
npm install strapi-plugin-menus@latest
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: ss-cms@0.1.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   peer react@"^18.2.0" from react-dom@18.2.0
npm ERR!   node_modules/react-dom
npm ERR!     peer react-dom@">= 16.8.0" from styled-components@5.3.10
npm ERR!     node_modules/styled-components
npm ERR!       peer styled-components@"^5.3.3" from @strapi/helper-plugin@4.10.5
npm ERR!       node_modules/@strapi/helper-plugin
npm ERR!         peer @strapi/helper-plugin@"^4.9.0" from strapi-plugin-menus@1.4.1
npm ERR!         node_modules/strapi-plugin-menus
npm ERR!       1 more (@strapi/design-system)
npm ERR!     peer react-dom@"^17.0.0 || ^18.0.0" from @strapi/design-system@1.7.5
npm ERR!     node_modules/@strapi/design-system
npm ERR!       peer @strapi/design-system@"^1.6.6" from strapi-plugin-menus@1.4.1
npm ERR!       node_modules/strapi-plugin-menus
npm ERR!         strapi-plugin-menus@"1.4.1" from the root project
npm ERR!       1 more (@strapi/helper-plugin)
npm ERR!     1 more (@strapi/icons)
npm ERR!   peer react@">=15" from react-router-dom@5.3.4
npm ERR!   node_modules/react-router-dom
npm ERR!     peer react-router-dom@"^5.3.4" from @strapi/helper-plugin@4.10.5
npm ERR!     node_modules/@strapi/helper-plugin
npm ERR!       peer @strapi/helper-plugin@"^4.9.0" from strapi-plugin-menus@1.4.1
npm ERR!       node_modules/strapi-plugin-menus
npm ERR!         strapi-plugin-menus@"1.4.1" from the root project
npm ERR!     peer react-router-dom@"^5.2.0" from @strapi/design-system@1.7.5
npm ERR!     node_modules/@strapi/design-system
npm ERR!       peer @strapi/design-system@"^1.6.6" from strapi-plugin-menus@1.4.1
npm ERR!       node_modules/strapi-plugin-menus
npm ERR!         strapi-plugin-menus@"1.4.1" from the root project
npm ERR!       1 more (@strapi/helper-plugin)
npm ERR!   3 more (styled-components, @strapi/design-system, @strapi/icons)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^17.0.2" from @strapi/helper-plugin@4.10.5
npm ERR! node_modules/@strapi/helper-plugin
npm ERR!   peer @strapi/helper-plugin@"^4.9.0" from strapi-plugin-menus@1.4.1
npm ERR!   node_modules/strapi-plugin-menus
npm ERR!     strapi-plugin-menus@"1.4.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/ansis/.npm/_logs/2023-05-22T17_06_03_841Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ansis/.npm/_logs/2023-05-22T17_06_03_841Z-debug-0.log
```

And here is my package.json:
```json
{
  "name": "strapi",
  "private": true,
  "version": "0.1.0",
  "description": "A Strapi application",
  "scripts": {
    "develop": "strapi develop",
    "start": "strapi start",
    "build": "strapi build",
    "strapi": "strapi"
  },
  "dependencies": {
    "@_sh/strapi-plugin-ckeditor": "^2.0.4",
    "@retikolo/drag-drop-content-types": "^1.3.4",
    "@strapi/plugin-i18n": "4.6.1",
    "@strapi/plugin-users-permissions": "4.6.1",
    "@strapi/strapi": "4.6.1",
    "better-sqlite3": "8.0.1"
  },
  "author": {
    "name": "A Strapi developer"
  },
  "strapi": {
    "uuid": "dfe2bc62-7252-40f5-9d15-1e36d30b7a0a"
  },
  "engines": {
    "node": ">=14.19.1 <=18.x.x",
    "npm": ">=6.0.0"
  },
  "license": "MIT"
}

```

I literally have no idea where to even start looking to fix this. Anyone with more experience in npm dependencies?

**Comment by Toolbox-AI** [comment]
Its because the plugin says it only works with React 17.xxx but you have installed react 18.2

I have the same problem and came here to check if it is compatible. (Force)

But i guess i will just try anyways. If anyone encounters any Bugs let me know before i move to production.

**Comment by johnny-prescott** [comment]
will install on --force for me but not run. Develop just will exit during init stage.

**Comment by Toolbox-AI** [comment]
Worked so far on me

**Comment by mattmilburn** [comment]
Hi @Ansis100 @johnny-prescott @Toolbox-AI This issue should be resolved in the latest version v4.11.x of Strapi. The plugin itself also updated today but these kinds of errors is not something we can fix through the plugin.

Some Strapi releases actually had dependency conflicts through the helper plugin and design system and we had to wait for the next Strapi update in order to upgrade without issues.
